Fix new lint errors

```
^
106
Error: pkg/apis/workflow/v1alpha1/workflow_types.go:1310:7: S1002: should omit comparison to bool constant, can be simplified to `match` (gosimple)
107
			if match == true {
108
			   ^
109
Error: test/e2e/cli_test.go:1344:32: S1039: unnecessary use of fmt.Sprintf (gosimple)
110
				assert.Contains(t, output, fmt.Sprintf("Created \"main.log\""))
111
				                           ^
112
Error: test/e2e/cli_test.go:1345:32: S1039: unnecessary use of fmt.Sprintf (gosimple)
113
				assert.Contains(t, output, fmt.Sprintf("Created \"hello_world.tgz\""))
114
				                           ^
115
Error: test/e2e/cli_test.go:1346:32: S1039: unnecessary use of fmt.Sprintf (gosimple)
116
				assert.Contains(t, output, fmt.Sprintf("Created \"bye_world.tgz\""))
117
				                           ^
118
Error: workflow/artifacts/raw/raw_test.go:48:2: SA5001: should check returned error before deferring rc.Close() (staticcheck)
[119](https://github.com/argoproj/argo-workflows/runs/6357156287?check_suite_focus=true#step:5:119)
	defer rc.Close()
[120](https://github.com/argoproj/argo-workflows/runs/6357156287?check_suite_focus=true#step:5:120)
	^
```

--- a/pkg/apis/workflow/v1alpha1/workflow_types.go
+++ b/pkg/apis/workflow/v1alpha1/workflow_types.go
@@ -1273,7 +1273,7 @@ func NewArtifactSearchQuery() *ArtifactSearchQuery {
 
 func (q *ArtifactSearchQuery) anyArtifactGCStrategy() bool {
 	for _, val := range q.ArtifactGCStrategies {
-		if val == true {
+		if val {
 			return val
 		}
 	}
@@ -1307,7 +1307,7 @@ func (w *Workflow) SearchArtifacts(q *ArtifactSearchQuery) ArtifactSearchResults
 			if q.NodeId != "" && n.ID != q.NodeId {
 				match = false
 			}
-			if match == true {
+			if match {
 				results = append(results, ArtifactSearchResult{Artifact: a, NodeID: n.ID})
 			}
 		}

--- a/test/e2e/cli_test.go
+++ b/test/e2e/cli_test.go
@@ -1341,9 +1341,9 @@ func (s *CLISuite) TestWorkflowCopyArtifact() {
 		Given().
 		RunCli([]string{"cp", "@latest", "outputDir"}, func(t *testing.T, output string, err error) {
 			if assert.NoError(t, err) {
-				assert.Contains(t, output, fmt.Sprintf("Created \"main.log\""))
-				assert.Contains(t, output, fmt.Sprintf("Created \"hello_world.tgz\""))
-				assert.Contains(t, output, fmt.Sprintf("Created \"bye_world.tgz\""))
+				assert.Contains(t, output, "Created \"main.log\"")
+				assert.Contains(t, output, "Created \"hello_world.tgz\"")
+				assert.Contains(t, output, "Created \"bye_world.tgz\"")
 			}
 		})
 	os.RemoveAll("outputDir")
@@ -1356,9 +1356,9 @@ func (s *CLISuite) TestWorkflowCopyArtifact() {
 		Given().
 		RunCli([]string{"cp", "@latest", "outputDir", "--template-name", "bye"}, func(t *testing.T, output string, err error) {
 			if assert.NoError(t, err) {
-				assert.Contains(t, output, fmt.Sprintf("Created \"main.log\""))
-				assert.Contains(t, output, fmt.Sprintf("Created \"bye_world.tgz\""))
-				assert.NotContains(t, output, fmt.Sprintf("Created \"hello_world.tgz\""))
+				assert.Contains(t, output, "Created \"main.log\"")
+				assert.Contains(t, output, "Created \"bye_world.tgz\"")
+				assert.NotContains(t, output, "Created \"hello_world.tgz\"")
 			}
 		})
 	os.RemoveAll("outputDir")
@@ -1371,9 +1371,9 @@ func (s *CLISuite) TestWorkflowCopyArtifact() {
 		Given().
 		RunCli([]string{"cp", "@latest", "outputDir", "--artifact-name", "hello_world"}, func(t *testing.T, output string, err error) {
 			if assert.NoError(t, err) {
-				assert.NotContains(t, output, fmt.Sprintf("Created \"main.log\""))
-				assert.NotContains(t, output, fmt.Sprintf("Created \"bye_world.tgz\""))
-				assert.Contains(t, output, fmt.Sprintf("Created \"hello_world.tgz\""))
+				assert.NotContains(t, output, "Created \"main.log\"")
+				assert.NotContains(t, output, "Created \"bye_world.tgz\"")
+				assert.Contains(t, output, "Created \"hello_world.tgz\"")
 			}
 		})
 	os.RemoveAll("outputDir")
@@ -1387,9 +1387,9 @@ func (s *CLISuite) TestWorkflowCopyArtifact() {
 		RunCli([]string{"cp", "@latest", ".", "--path", "/{templateName}/{artifactName}/"}, func(t *testing.T, output string, err error) {
 			if assert.NoError(t, err) {
 				//Assert everything was stored
-				assert.Contains(t, output, fmt.Sprintf("Created \"main.log\""))
-				assert.Contains(t, output, fmt.Sprintf("Created \"bye_world.tgz\""))
-				assert.Contains(t, output, fmt.Sprintf("Created \"hello_world.tgz\""))
+				assert.Contains(t, output, "Created \"main.log\"")
+				assert.Contains(t, output, "Created \"bye_world.tgz\"")
+				assert.Contains(t, output, "Created \"hello_world.tgz\"")
 				//Assert filepaths are correct
 				statStrip := func(f os.FileInfo, err error) error {
 					return err

--- a/workflow/artifacts/raw/raw_test.go
+++ b/workflow/artifacts/raw/raw_test.go
@@ -45,8 +45,8 @@ func TestOpenStream(t *testing.T) {
 	}
 	driver := &raw.ArtifactDriver{}
 	rc, err := driver.OpenStream(art)
-	defer rc.Close()
 	assert.NoError(t, err)
+	defer rc.Close()
 
 	dat, err := io.ReadAll(rc)
 	assert.NoError(t, err)

